tests/buffer/buffer_api.cpp

```diff
@@ -317,6 +424,34 @@ class TEST_NAME : public util::test_base {
     set_test_info(out, TOSTRING(TEST_NAME), TEST_FILE);
   }
 
+  /**
+   * @brief Tests reinterpreting a buffer
+   * @tparam T Underlying data type of the input buffer
+   * @tparam numDims Number of input dimensions
+   * @param log Logger object
+   */
+  template <typename T, int numDims>
+  void test_type_reinterpret(util::logger& log) {
+    static constexpr size_t inputElemsPerDim = 4;
+    cl::sycl::vector_class<uint8_t> reinterpretInputData(
+        sizeof(T) * inputElemsPerDim * numDims);
+    using ReinterpretT = flip_signedness_t<T>;
+
+    // Check reinterpreting with a range
+    static constexpr auto numElems = inputElemsPerDim * numDims;
+    test_buffer_reinterpret<uint8_t, T>{sizeof(T) * numElems, numElems}
+        .template check<numDims>(reinterpretInputData.data(), log);
+
+    // Check reinterpreting without a range to 1D
+    test_buffer_reinterpret_no_range<uint8_t, T, numDims, 1>::check(
```

Comment:
To keep these tests working for SYCL-1.2.1 implementations we can surround SYCL-2020 API with
SYCL_LANGUAGE_VERSION check.

Reply:
Is there SYCL 2020 code in this CTS?

Reply:
This feature made it into 2020 provisional, so I've added a check for `SYCL_LANGUAGE_VERSION >= 202001`